File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/BlueBomb.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class BlueBomb extends Equipable {
	{
		name = Locale.get("blue_bomb");
		description = Locale.get("blue_bomb_desc");
		sprite = "item-mana_bombs";
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).manaBombs = true;
			((Player) this.owner).manaRegenRate += 0.2f;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).manaBombs = false;
			((Player) this.owner).manaRegenRate -= 0.2f;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/BlueBomb.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class BlueBomb : Equippable() {
	init {
		name = Locale.get("blue_bomb")
		description = Locale.get("blue_bomb_desc")
		sprite = "item-mana_bombs"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)

		this.owner.manaBombs = true
		this.owner.manaRegenRate += 0.5f
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)

		this.owner.manaBombs = false
		this.owner.manaRegenRate -= 0.5f
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}
}